app/src/pages/inside/dashboardPage/widgets/projectActivity/activities/analysisProperties.jsx

```diff
@@ -0,0 +1,122 @@
+import React, { Component, Fragment } from 'react';
+import PropTypes from 'prop-types';
+import Parser from 'html-react-parser';
+import classNames from 'classnames/bind';
+import { injectIntl, intlShape, defineMessages, FormattedMessage } from 'react-intl';
+import { ANALYSIS } from 'common/constants/settingsTabs';
+import styles from './common.scss';
+
+const cx = classNames.bind(styles);
+
+const messages = defineMessages({
+  auto_analyze: {
+    id: 'UpdateAnalysisSettings.autoAnalyze',
+    defaultMessage: 'switch Auto Analysis',
+  },
+  number_of_log_lines: {
+    id: 'UpdateAnalysisSettings.numberOfLogLines',
+    defaultMessage: 'Number of log lines',
+  },
+  min_doc_freq: {
+    id: 'UpdateAnalysisSettings.minDocFreq',
+    defaultMessage: 'Minimum document frequency',
+  },
+  min_should_match: {
+    id: 'UpdateAnalysisSettings.minShouldMatch',
+    defaultMessage: 'Minimum should match',
+  },
+  min_term_freq: {
+    id: 'UpdateAnalysisSettings.minTermFreq',
+    defaultMessage: 'Minimum term frequency',
+  },
+  analyze_mode: {
+    id: 'UpdateAnalysisSettings.analyzeMode',
+    defaultMessage: 'Base for Auto Analysis',
+  },
+  from: {
+    id: 'AnalysisProperties.from',
+    defaultMessage: 'from',
+  },
+  to: {
+    id: 'AnalysisProperties.to',
+    defaultMessage: 'to',
+  },
+  all: {
+    id: 'AnalysisProperties.all',
+    defaultMessage: 'All',
+  },
+  on: {
+    id: 'AnalysisProperties.on',
+    defaultMessage: 'ON',
+  },
+  off: {
+    id: 'AnalysisProperties.off',
+    defaultMessage: 'OFF',
+  },
+});
+
+@injectIntl
+export class AnalysisProperties extends Component {
+  static propTypes = {
+    intl: intlShape.isRequired,
+    activity: PropTypes.object,
+  };
+  static defaultProps = {
+    activity: {},
+  };
+
+  getActivityHistory = (activity) => {
+    const size = Object.keys(activity.history).length;
+    let i = 0;
+    let activities = '';
+    Object.keys(activity.history).forEach((key) => {
+      i += 1;
```

Comment:
But you have `index` variable in second argument of callback function. It isn't the same?